docs/proposals/going-towards-to-crd.md

```diff
@@ -0,0 +1,104 @@
+
+In the master branch, we have Service Catalog with the CRD approach. New releases are created with the `0.2.x` version. On master branch, we are doing both the bug fixing and adding new features.
+
+Such strategy will help us to get new customers cause we will show that the CRD is our new direction on which we want to focus. 
+ 
+On the other hand, existing Service Catalog customers will see that the old solution is deprecated and the new features will be available only in the new approach. Thanks to that they will consider updating their system to the newest version as soon as possible cause the goal is to use the CRD solution as Kubernetes community recommends that.
+
+We can set a reasonable time for supporting bug fixing for the Service Catalog with Aggregated API, e.g. 1 year.
+
+With such solution, problems described in the previous section are mitigated.  
+     
+## Migration Support
+
+If we want to convince existing customers to use the new Service Catalog version then we need to create a migration guide/scripts. The migration needs to be as simple as possible.
+
+The migration will be simpler when we will hardcode for a given functionality in Aggregated API. 
+
+#### Details
+
+The migration logic can be placed in the Service Catalog Helm Chart. 
+When someone wants to upgrade existing Service Catalog to the newest one, then he can execute `helm upgrade {release-name} svc-cat/catalog`
+
+Raw scenario:
+- **pre-upgrade hook**
+  - replace the api-server deployment with the api-server that has a read-only mode
+  - backup the Service Catalog resources to the persistent volume
+- **upgrade**
+  - removes the api-server
+  - remove the etcd storage
+  - adjust secrets, RBAC etc.
+  - upgrade controller manager
+  - install webhook server
+- **post-upgrade**
+  - scale down the controller manager to 0  
+  - restore the Service Catalog resources - Spec and Status (status is important cause we do not want to trigger provisioning for already process items)
+  - scale up controller manager
+
+
+## Sum up
+
+We need to be sure that customer will exactly know in which direction we want to go. Supporting only bug fixing for api-server could simplify adding new features while we will still provide support for older clients. The migration process will also be easier when we will hardcode for a given functionality in Aggregated API. For new contributors, it will be much easier to get familiar with the code base where only one approach is used. 
+
+From our perspective having both approaches at the same time and supporting bug fixing and feature for both of them causing only that we are postponing the whole process. Sooner or later, we will have to take all these steps and back to the discussion. The only problem with doing that later can be that it will be harder (technical debt, confused customers, etc.)
```

Comment:
```suggestion
From our perspective supporting both approaches at the same time with bug fixing and features will just postpone the whole process. Sooner or later, we will have to take all these steps and back to the discussion. The only problem with doing that later can be that it will be harder (technical debt, confused customers, etc.)

What we need:
- set a date e.g. Jan 2020 when api-server will be erased from Service Catalog repository 
- as soon as possible communicate via service catalog sig and all channels about above decision, provide CRD solution and migration guide already.
```